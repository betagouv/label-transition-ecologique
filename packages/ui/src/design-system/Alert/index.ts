export * from './Alert';
export * from './BottomAlertOkCancel';
