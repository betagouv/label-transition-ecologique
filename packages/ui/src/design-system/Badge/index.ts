export * from './Badge';
export * from './utils';
