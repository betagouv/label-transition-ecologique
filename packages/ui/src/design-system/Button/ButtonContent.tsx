import {Icon, IconSize} from '@design-system/Icon';
import {buttonThemeClassnames} from './theme';
import {ButtonContentProps, ButtonSize} from './types';

const getIconSize = (size: ButtonSize): IconSize | undefined => {
  // Les sizes des icÃ´nes ne matchent pas celles des boutons
  switch (size) {
    case 'xs':
      return 'sm';
    case 'sm':
      return 'md';
    case 'md':
      return 'md';
    case 'xl':
      return 'lg';
  }
};

/** Affiche le contenu d'un bouton */
const ButtonContent = ({
  variant = 'primary',
  size = 'md',
  icon,
  children,
  disabled,
}: ButtonContentProps) => {
  const buttonState = disabled ? 'disabled' : 'default';

  return (
    <>
      {!!icon && (
        <Icon
          icon={icon}
          size={getIconSize(size)}
          className={buttonThemeClassnames[variant][buttonState].icon}
        />
      )}

      {!!children && children}
    </>
  );
};

export default ButtonContent;
