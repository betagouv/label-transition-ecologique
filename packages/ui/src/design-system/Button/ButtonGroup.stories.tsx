import {useState} from 'react';
import {Meta, StoryObj} from '@storybook/react';

import {ButtonGroup} from './ButtonGroup';

const meta: Meta<typeof ButtonGroup> = {
  component: ButtonGroup,
  args: {
    activeButtonId: '2',
    buttons: [
      {id: '1', children: 'Bouton 1'},
      {id: '2', children: 'Bouton 2'},
      {id: '3', children: 'Bouton 3'},
    ],
  },
  render: args => {
    const [activeId, setActiveId] = useState(args.activeButtonId);
    return (
      <ButtonGroup
        {...args}
        activeButtonId={activeId}
        buttons={args.buttons.map(button => ({
          ...button,
          onClick: () => setActiveId(button.id),
        }))}
      />
    );
  },
};

export default meta;

type Story = StoryObj<typeof ButtonGroup>;

export const Default: Story = {
  args: {},
};

export const Size: Story = {
  args: {
    size: 'xs',
  },
};

export const Disabled: Story = {
  args: {
    buttons: [
      {id: '1', children: 'Bouton 1', disabled: true},
      {id: '2', children: 'Bouton 2'},
      {id: '3', children: 'Bouton 3'},
    ],
  },
};

export const Deux: Story = {
  args: {
    buttons: [
      {id: '1', children: 'Bouton 1'},
      {id: '2', children: 'Bouton 2'},
    ],
  },
};
