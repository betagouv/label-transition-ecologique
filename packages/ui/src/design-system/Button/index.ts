export * from './Button';
export * from './NotificationButton';
