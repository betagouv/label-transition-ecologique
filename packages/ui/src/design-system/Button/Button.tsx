import {Ref, forwardRef} from 'react';
import classNames from 'classnames';

import {ButtonHTMLProps, isAnchor} from 'utils/types';
import {buttonSizeClassnames, buttonThemeClassnames} from './theme';
import ButtonContent from './ButtonContent';
import {ButtonContentProps, ButtonProps} from './types';

/**
 * Composant bouton par défaut, ayant pour props toutes les props habituelles d'un button tag.
 * */
// On déstructure toutes les props rajoutées qui ne sont pas des props des tags HTML <button> ou <a>
// Ce qui nous permet de ne donner que les props restantes natives au tag HTML
export const Button = forwardRef(
  (
    {
      children,
      disabled = false,
      variant = 'primary',
      size = 'md',
      className,
      icon,
      iconPosition,
      external,
      ...props
    }: ButtonProps,
    ref?: Ref<HTMLButtonElement | HTMLAnchorElement>
  ) => {
    const isIconButton = !children;

    const buttonState = disabled ? 'disabled' : 'default';

    const {text, background, border} =
      buttonThemeClassnames[variant][buttonState];

    const sizeClassName =
      buttonSizeClassnames[size][isIconButton ? 'iconButton' : 'textButton'];

    const buttonClassname = classNames(
      'w-fit flex items-center border-solid group',
      {
        // Layout du bouton
        'gap-1': size === 'xs' || (size === 'sm' && variant === 'underlined'),
        'gap-2':
          (size === 'sm' && variant !== 'underlined') ||
          size === 'md' ||
          size === 'xl',
        // Styles du curseur
        'cursor-pointer': !disabled,
        'cursor-not-allowed': disabled,
        // Bordures et polices
        'rounded-lg border font-bold': variant !== 'underlined',
        'border-b border-t-0 border-x-0 !p-px font-medium':
          variant === 'underlined',
        'hover:border-b-2 hover:!pb-0': variant === 'underlined' && !disabled,
      },
      text,
      background,
      border,
      sizeClassName
    );

    /** Reconstitution des props données au contenu du bouton */
    const buttonContentProps: ButtonContentProps = {
      children,
      variant,
      size,
      icon,
      disabled,
    };

    const isButton = !isAnchor(props);

    /** On affiche un bouton par défaut */
    if (isButton) {
      // On réintegre la prop disabled qui a été déstructurée plus haut
      const buttonProps = {...props, disabled} as ButtonHTMLProps;
      return (
        <button
          ref={ref as Ref<HTMLButtonElement>}
          {...buttonProps}
          className={classNames(
            buttonClassname,
            {'flex-row-reverse': iconPosition === 'right'},
            className
          )}
          onClick={evt => {
            evt.stopPropagation();
            buttonProps.onClick?.(evt);
          }}
        >
          <ButtonContent {...buttonContentProps} />
        </button>
      );

      /** Ou bien une ancre si un lien href est donné (cf ./types.ts) */
    } else {
      const anchorProps = props;
      const openInNewTab = external || props.target === '_blank';
      return (
        <a
          {...anchorProps}
          ref={ref as Ref<HTMLAnchorElement>}
          // bg-none permet d'effacer un style dsfr appliqué à la balise <a/>
          // after:hidden supprime l'icône external par défaut du dsfr
          className={classNames(
            'bg-none after:hidden',
            buttonClassname,
            {
              'flex-row-reverse':
                iconPosition === 'right' ||
                (openInNewTab && iconPosition !== 'left'),
            },
            className
          )}
          target={openInNewTab ? '_blank' : anchorProps.target}
          rel={openInNewTab ? 'noreferrer noopener' : anchorProps.rel}
          onClick={evt => {
            evt.stopPropagation();
            if (disabled) evt.preventDefault();
            else anchorProps.onClick?.(evt);
          }}
        >
          <ButtonContent
            {...buttonContentProps}
            icon={openInNewTab ? 'external-link-line' : icon}
          />
        </a>
      );
    }
  }
);
