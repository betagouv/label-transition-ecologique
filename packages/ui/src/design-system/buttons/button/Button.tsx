import classNames from 'classnames';
import {buttonThemeClassnames} from '../utils';
import ButtonContent from './ButtonContent';
import {ButtonProps, isAnchorButton, isDefaultButton} from '../types';

/**
 * Composant bouton par défaut, ayant pour props toutes les props habituelles d'un button tag.
 * */
export const Button = (props: ButtonProps) => {
  const {
    children,
    disabled = false,
    variant = 'primary',
    size = 'md',
    className,
  } = props;

  const isIconButton = !children;
  const {text, background, border} = buttonThemeClassnames[variant];

  const buttonClassname = classNames(
    'rounded-lg w-fit border border-solid font-bold group',
    {
      // Format du bouton en fonction de la taille choisie
      'text-xs py-2': size === 'xs',
      'text-sm py-3': size === 'sm',
      'text-base py-3.5': size === 'md',
      'text-2xl py-5': size === 'xl',

      // Padding en fonction de la taille choisie
      'px-5':
        (size === 'xs' && !isIconButton) || (size === 'xl' && isIconButton),
      'px-6': size !== 'xs' && !isIconButton,

      // Padding sur les boutons ne contenant qu'une icône
      'px-2': size === 'xs' && isIconButton,
      'px-3': size === 'sm' && isIconButton,
      'px-3.5': size === 'md' && isIconButton,

      // Styles du curseur
      'cursor-pointer': !disabled,
      'cursor-not-allowed': disabled,
    },
    text,
    background,
    border
  );

  if (isDefaultButton(props)) {
    return (
      <button {...props} className={classNames(buttonClassname, className)}>
        <ButtonContent {...{...props, children, disabled, variant, size}} />
      </button>
    );
  } else if (isAnchorButton(props)) {
    const {external = false} = props;

    return (
      <a
        {...props}
        // bg-none permet d'effacer un style dsfr appliqué à la balise <a/>
        // after:hidden supprime l'icône external par défaut du dsfr
        className={classNames(
          'inline-block w-fit bg-none after:hidden',
          buttonClassname,
          className
        )}
        target={external ? '_blank' : props.target}
        rel={external ? 'noreferrer noopener' : props.rel}
        onClick={evt => {
          if (disabled) evt.preventDefault();
        }}
      >
        <ButtonContent
          {...{...props, children, disabled, variant, size}}
          external={external || props.target === '_blank'}
        />
      </a>
    );
  }
};
