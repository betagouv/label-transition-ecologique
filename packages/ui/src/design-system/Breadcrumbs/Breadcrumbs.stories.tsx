import { Button } from '@/ui/design-system/Button';
import { Meta, StoryObj } from '@storybook/react';
import { useState } from 'react';
import { Breadcrumbs } from './Breadcrumbs';

const meta: Meta<typeof Breadcrumbs> = {
  component: Breadcrumbs,
  args: {
    items: [
      { label: 'Plan Vélo 2020 - 2024' },
      {
        label:
          '0.0 Etudions un scénario de rupture pour atteindre la neutralité carbone en 2050',
      },
      { label: 'Etudions un scénario de rupture pour la période 2030-2050' },
    ],
  },
};

const buttons = [
  { label: 'Plan Vélo 2020 - 2024' },
  {
    label:
      '0.0 Etudions un scénario de rupture pour atteindre la neutralité carbone en 2050',
  },
  { label: 'Etudions un scénario de rupture pour la période 2030-2050' },
];

export default meta;

type Story = StoryObj<typeof Breadcrumbs>;

export const Default: Story = {
  render: () => {
    const [currentButtons, setCurrentButtons] = useState(buttons);
    return (
      <div className="flex flex-col gap-8">
        <Breadcrumbs
          items={currentButtons}
          onClick={(index: number) =>
            setCurrentButtons((prevState) => prevState.slice(0, index + 1))
          }
        />

        <Button onClick={() => setCurrentButtons(buttons)} size="xs">
          Reset
        </Button>
      </div>
    );
  },
};

export const Sizes: Story = {
  render: () => {
    const [currentButtons, setCurrentButtons] = useState(buttons);
    return (
      <div className="flex flex-col gap-8">
        <Breadcrumbs
          size="xs"
          items={currentButtons}
          onClick={(index: number) =>
            setCurrentButtons((prevState) => prevState.slice(0, index + 1))
          }
        />
        <Breadcrumbs
          size="sm"
          items={currentButtons}
          onClick={(index: number) =>
            setCurrentButtons((prevState) => prevState.slice(0, index + 1))
          }
        />
        <Breadcrumbs
          size="md"
          items={currentButtons}
          onClick={(index: number) =>
            setCurrentButtons((prevState) => prevState.slice(0, index + 1))
          }
        />
        <Breadcrumbs
          size="xl"
          items={currentButtons}
          onClick={(index: number) =>
            setCurrentButtons((prevState) => prevState.slice(0, index + 1))
          }
        />

        <Button onClick={() => setCurrentButtons(buttons)} size="xs">
          Reset
        </Button>
      </div>
    );
  },
};

export const ReadOnlyMode: Story = {
  render: () => {
    return (
      <div className="flex flex-col gap-8">
        <Breadcrumbs items={buttons} />
      </div>
    );
  },
};
