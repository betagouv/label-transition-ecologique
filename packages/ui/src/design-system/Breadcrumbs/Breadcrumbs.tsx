import {Button, ButtonSize} from '@design-system/Button';
import {Icon, IconSize} from '@design-system/Icon';
import classNames from 'classnames';

const buttonSizeToIconSize: Record<ButtonSize, IconSize> = {
  xs: 'sm',
  sm: 'md',
  md: 'lg',
  xl: '2xl',
};

type BreadcrumbsProps = {
  /** Liste des boutons à afficher dans le fil d'ariane */
  buttons?: {
    label: string;
    href?: string;
    onClick?: () => void;
  }[];
  /** Click détecté sur un des boutons, avec envoi de son index */
  onClick?: (index: number) => void;
  /** Taille des boutons */
  size?: ButtonSize;
};

/**
 * Fil d'ariane, avec liens ou boutons
 */
export const Breadcrumbs = ({
  buttons,
  size = 'md',
  onClick,
}: BreadcrumbsProps) => {
  const readOnlyMode =
    buttons.filter(button => button.href || button.onClick).length === 0 &&
    !onClick;

  return (
    <div className="flex flex-wrap gap-x-1 gap-y-0.5">
      {(buttons ?? []).map((button, index) => {
        const isLastElement = index === buttons.length - 1;

        return (
          <div
            key={button.label}
            className="flex items-center shrink-0 gap-x-1"
          >
            <Button
              disabled={isLastElement && !readOnlyMode}
              className={classNames(
                'font-normal border-none hover:!pb-px text-left',
                {
                  '!cursor-default': isLastElement || readOnlyMode,
                  '!text-primary-9': isLastElement && !readOnlyMode,
                  '!text-grey-6': !isLastElement || readOnlyMode,
                  'hover:!text-primary-9 transition':
                    !isLastElement && !readOnlyMode,
                }
              )}
              variant="underlined"
              size={size}
              onClick={() => {
                onClick(index);
                button.onClick();
              }}
              href={button.href}
            >
              {button.label}
            </Button>

            {/* Icône `>` entre chaque bouton */}
            {!isLastElement && (
              <Icon
                icon="arrow-right-s-line"
                size={buttonSizeToIconSize[size]}
                className="text-grey-8"
              />
            )}
          </div>
        );
      })}
    </div>
  );
};
