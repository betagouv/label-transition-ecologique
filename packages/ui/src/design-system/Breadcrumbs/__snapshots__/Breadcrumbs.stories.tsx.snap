// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`design-system/Breadcrumbs Default smoke-test 1`] = `
<div class="flex flex-col gap-8">
  <div class="flex flex-wrap gap-x-1 gap-y-0.5">
    <div class="flex items-center shrink-0 gap-x-1">
      <button class="w-fit flex items-center border-solid group gap-2 cursor-pointer border-b border-t-0 border-x-0 !p-px font-medium hover:border-b-2 hover:!pb-0 text-primary-8 !bg-transparent border-primary-7 text-base py-3 px-6 font-normal border-none hover:!pb-px text-left !text-grey-6 hover:!text-primary-9 transition">
        Plan Vélo 2020 - 2024
      </button>
      <span class="ri-arrow-right-s-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] text-grey-8">
      </span>
    </div>
    <div class="flex items-center shrink-0 gap-x-1">
      <button class="w-fit flex items-center border-solid group gap-2 cursor-pointer border-b border-t-0 border-x-0 !p-px font-medium hover:border-b-2 hover:!pb-0 text-primary-8 !bg-transparent border-primary-7 text-base py-3 px-6 font-normal border-none hover:!pb-px text-left !text-grey-6 hover:!text-primary-9 transition">
        0.0 Etudions un scénario de rupture pour atteindre la neutralité carbone en 2050
      </button>
      <span class="ri-arrow-right-s-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] text-grey-8">
      </span>
    </div>
    <div class="flex items-center shrink-0 gap-x-1">
      <button disabled
              class="w-fit flex items-center border-solid group gap-2 cursor-not-allowed border-b border-t-0 border-x-0 !p-px font-medium !text-primary-3 bg-transparent border-primary-3 text-base py-3 px-6 font-normal border-none hover:!pb-px text-left !cursor-default !text-primary-9"
      >
        Etudions un scénario de rupture pour la période 2030-2050
      </button>
    </div>
  </div>
  <button class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5">
    Reset
  </button>
</div>
`;

exports[`design-system/Breadcrumbs Sizes smoke-test 1`] = `
<div class="flex flex-col gap-8">
  <div class="flex flex-wrap gap-x-1 gap-y-0.5">
    <div class="flex items-center shrink-0 gap-x-1">
      <button class="w-fit flex items-center border-solid group gap-1 cursor-pointer border-b border-t-0 border-x-0 !p-px font-medium hover:border-b-2 hover:!pb-0 text-primary-8 !bg-transparent border-primary-7 text-xs py-2 px-5 font-normal border-none hover:!pb-px text-left !text-grey-6 hover:!text-primary-9 transition">
        Plan Vélo 2020 - 2024
      </button>
      <span class="ri-arrow-right-s-line font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] text-grey-8">
      </span>
    </div>
    <div class="flex items-center shrink-0 gap-x-1">
      <button class="w-fit flex items-center border-solid group gap-1 cursor-pointer border-b border-t-0 border-x-0 !p-px font-medium hover:border-b-2 hover:!pb-0 text-primary-8 !bg-transparent border-primary-7 text-xs py-2 px-5 font-normal border-none hover:!pb-px text-left !text-grey-6 hover:!text-primary-9 transition">
        0.0 Etudions un scénario de rupture pour atteindre la neutralité carbone en 2050
      </button>
      <span class="ri-arrow-right-s-line font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] text-grey-8">
      </span>
    </div>
    <div class="flex items-center shrink-0 gap-x-1">
      <button disabled
              class="w-fit flex items-center border-solid group gap-1 cursor-not-allowed border-b border-t-0 border-x-0 !p-px font-medium !text-primary-3 bg-transparent border-primary-3 text-xs py-2 px-5 font-normal border-none hover:!pb-px text-left !cursor-default !text-primary-9"
      >
        Etudions un scénario de rupture pour la période 2030-2050
      </button>
    </div>
  </div>
  <div class="flex flex-wrap gap-x-1 gap-y-0.5">
    <div class="flex items-center shrink-0 gap-x-1">
      <button class="w-fit flex items-center border-solid group gap-1 cursor-pointer border-b border-t-0 border-x-0 !p-px font-medium hover:border-b-2 hover:!pb-0 text-primary-8 !bg-transparent border-primary-7 text-sm py-2.5 px-6 font-normal border-none hover:!pb-px text-left !text-grey-6 hover:!text-primary-9 transition">
        Plan Vélo 2020 - 2024
      </button>
      <span class="ri-arrow-right-s-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] text-grey-8">
      </span>
    </div>
    <div class="flex items-center shrink-0 gap-x-1">
      <button class="w-fit flex items-center border-solid group gap-1 cursor-pointer border-b border-t-0 border-x-0 !p-px font-medium hover:border-b-2 hover:!pb-0 text-primary-8 !bg-transparent border-primary-7 text-sm py-2.5 px-6 font-normal border-none hover:!pb-px text-left !text-grey-6 hover:!text-primary-9 transition">
        0.0 Etudions un scénario de rupture pour atteindre la neutralité carbone en 2050
      </button>
      <span class="ri-arrow-right-s-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] text-grey-8">
      </span>
    </div>
    <div class="flex items-center shrink-0 gap-x-1">
      <button disabled
              class="w-fit flex items-center border-solid group gap-1 cursor-not-allowed border-b border-t-0 border-x-0 !p-px font-medium !text-primary-3 bg-transparent border-primary-3 text-sm py-2.5 px-6 font-normal border-none hover:!pb-px text-left !cursor-default !text-primary-9"
      >
        Etudions un scénario de rupture pour la période 2030-2050
      </button>
    </div>
  </div>
  <div class="flex flex-wrap gap-x-1 gap-y-0.5">
    <div class="flex items-center shrink-0 gap-x-1">
      <button class="w-fit flex items-center border-solid group gap-2 cursor-pointer border-b border-t-0 border-x-0 !p-px font-medium hover:border-b-2 hover:!pb-0 text-primary-8 !bg-transparent border-primary-7 text-base py-3 px-6 font-normal border-none hover:!pb-px text-left !text-grey-6 hover:!text-primary-9 transition">
        Plan Vélo 2020 - 2024
      </button>
      <span class="ri-arrow-right-s-line font-normal text-center h-6 w-6 text-[1.5rem] leading-[1.5rem] text-grey-8">
      </span>
    </div>
    <div class="flex items-center shrink-0 gap-x-1">
      <button class="w-fit flex items-center border-solid group gap-2 cursor-pointer border-b border-t-0 border-x-0 !p-px font-medium hover:border-b-2 hover:!pb-0 text-primary-8 !bg-transparent border-primary-7 text-base py-3 px-6 font-normal border-none hover:!pb-px text-left !text-grey-6 hover:!text-primary-9 transition">
        0.0 Etudions un scénario de rupture pour atteindre la neutralité carbone en 2050
      </button>
      <span class="ri-arrow-right-s-line font-normal text-center h-6 w-6 text-[1.5rem] leading-[1.5rem] text-grey-8">
      </span>
    </div>
    <div class="flex items-center shrink-0 gap-x-1">
      <button disabled
              class="w-fit flex items-center border-solid group gap-2 cursor-not-allowed border-b border-t-0 border-x-0 !p-px font-medium !text-primary-3 bg-transparent border-primary-3 text-base py-3 px-6 font-normal border-none hover:!pb-px text-left !cursor-default !text-primary-9"
      >
        Etudions un scénario de rupture pour la période 2030-2050
      </button>
    </div>
  </div>
  <div class="flex flex-wrap gap-x-1 gap-y-0.5">
    <div class="flex items-center shrink-0 gap-x-1">
      <button class="w-fit flex items-center border-solid group gap-2 cursor-pointer border-b border-t-0 border-x-0 !p-px font-medium hover:border-b-2 hover:!pb-0 text-primary-8 !bg-transparent border-primary-7 text-2xl py-4 px-6 font-normal border-none hover:!pb-px text-left !text-grey-6 hover:!text-primary-9 transition">
        Plan Vélo 2020 - 2024
      </button>
      <span class="ri-arrow-right-s-line font-normal text-center h-8 w-8 text-[2rem] leading-[2rem] text-grey-8">
      </span>
    </div>
    <div class="flex items-center shrink-0 gap-x-1">
      <button class="w-fit flex items-center border-solid group gap-2 cursor-pointer border-b border-t-0 border-x-0 !p-px font-medium hover:border-b-2 hover:!pb-0 text-primary-8 !bg-transparent border-primary-7 text-2xl py-4 px-6 font-normal border-none hover:!pb-px text-left !text-grey-6 hover:!text-primary-9 transition">
        0.0 Etudions un scénario de rupture pour atteindre la neutralité carbone en 2050
      </button>
      <span class="ri-arrow-right-s-line font-normal text-center h-8 w-8 text-[2rem] leading-[2rem] text-grey-8">
      </span>
    </div>
    <div class="flex items-center shrink-0 gap-x-1">
      <button disabled
              class="w-fit flex items-center border-solid group gap-2 cursor-not-allowed border-b border-t-0 border-x-0 !p-px font-medium !text-primary-3 bg-transparent border-primary-3 text-2xl py-4 px-6 font-normal border-none hover:!pb-px text-left !cursor-default !text-primary-9"
      >
        Etudions un scénario de rupture pour la période 2030-2050
      </button>
    </div>
  </div>
  <button class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5">
    Reset
  </button>
</div>
`;
